### PR TITLE
{CI} Replace deprecated `assertRaisesRegexp` and `assertEquals`

### PR DESCRIPTION
**Pull request by bebound**
`assertRaisesRegexp` is removed in 3.12: https://github.com/Azure/azure-cli/pull/29504
`assertEquals` is removed in 3.12: https://github.com/Azure/azure-cli/pull/29515

Apply same changes in extension.

This PR only changes the test file, it should not affect user.

### CODE REVIEW

**Comment by yonzhan** [comment]
CI

File: src/apic-extension/azext_apic_extension/tests/latest/test_definition_commands.py [hunk 1]
```diff
@@ -310,4 +310,4 @@ def test_examples_export_specification(self):
             # Check the exported file exists
             assert os.path.exists(self.kwargs['filename'])
         finally:
-            os.remove(self.kwargs['filename'])
\ No newline at end of file
+            os.remove(self.kwargs['filename'])
```
Comment (Pan-Qi) [inline src/apic-extension/azext_apic_extension/tests/latest/test_definition_commands.py:313]: Missing ending line

File: src/authV2/azext_authV2/tests/latest/test_authV2_scenario.py [hunk 2]
```diff
@@ -153,4 +153,4 @@ def test_authV2_authclassic(self, resource_group):
             JMESPathCheck('issuer', 'https://issuer_url'),
             JMESPathCheck('facebookAppId', 'facebook_id')]).get_output_in_json()
 
-        self.assertIn('https://audience1', result['allowedAudiences'])
\ No newline at end of file
+        self.assertIn('https://audience1', result['allowedAudiences'])
```
Comment (Pan-Qi) [inline src/authV2/azext_authV2/tests/latest/test_authV2_scenario.py:156]: Missing ending line

File: src/azure-firewall/azext_firewall/tests/latest/test_azure_firewall_scenario.py [hunk 3]
```diff
@@ -1471,4 +1471,4 @@ def test_azure_policy_idps_profiles(self, resource_group, resource_group_locatio
         ])  
 
         # delete policy 2
-        self.cmd('network firewall policy delete -g {rg} --name {policy_name_2}')
\ No newline at end of file
+        self.cmd('network firewall policy delete -g {rg} --name {policy_name_2}')
```
Comment (Pan-Qi) [inline src/azure-firewall/azext_firewall/tests/latest/test_azure_firewall_scenario.py:1474]: Missing ending line

File: src/connectedk8s/azext_connectedk8s/tests/latest/test_connectedk8s_scenario.py [hunk 4]
```diff
@@ -874,4 +874,4 @@ def test_skipping_ssl_verification(self,resource_group):
         self.cmd('aks delete -g {rg} -n {managed_cluster_name} -y')
 
         # delete the kube config
-        os.remove("%s" % (_get_test_data_file(managed_cluster_name + '-config.yaml')))
\ No newline at end of file
+        os.remove("%s" % (_get_test_data_file(managed_cluster_name + '-config.yaml')))
```
Comment (Pan-Qi) [inline src/connectedk8s/azext_connectedk8s/tests/latest/test_connectedk8s_scenario.py:877]: Missing ending line

File: src/network-analytics/azext_network_analytics/tests/latest/test_network_analytics.py [hunk 5]
```diff
@@ -70,6 +70,6 @@ def test_data_product_e2e(self, resource_group, location, user_id="test@microsof
 
         #List DP
         dplistinrg = self.cmd('az network-analytics data-product list --resource-group {group}').get_output_in_json()
-        self.assertEquals(len(dplistinrg), 0, msg="Data product list is not empty after deletion")
+        self.assertEqual(len(dplistinrg), 0, msg="Data product list is not empty after deletion")
 
-    pass
\ No newline at end of file
+    pass
```
Comment (Pan-Qi) [inline src/network-analytics/azext_network_analytics/tests/latest/test_network_analytics.py:75]: Missing ending line

**Review by Pan-Qi — APPROVED** [review]
Missed some ending line of file, other than that LGTM

File: src/network-analytics/azext_network_analytics/tests/latest/test_network_analytics.py (hunk 5 above)
Comment (jiasli) [reply src/network-analytics/azext_network_analytics/tests/latest/test_network_analytics.py:75]: I don't understand. This old file is missing ending line, not the new file.

File: src/connectedk8s/azext_connectedk8s/tests/latest/test_connectedk8s_scenario.py [hunk 6]
```diff
@@ -461,6 +461,10 @@ def test_enable_disable_features(self, resource_group):
                 "connectedk8s disable-features -n {name} -g {rg} --features cluster-connect --kube-config \
                 {kubeconfig} --kube-context {managed_cluster_name}-admin -y"
             )
+        with self.assertRaisesRegex(CLIError, "Disabling 'cluster-connect' feature is not allowed when \
+'custom-locations' feature is enabled."):
+            self.cmd('connectedk8s disable-features -n {name} -g {rg} --features cluster-connect --kube-config \
+                {kubeconfig} --kube-context {managed_cluster_name}-admin -y')
```
Comment (jiasli) [inline src/connectedk8s/azext_connectedk8s/tests/latest/test_connectedk8s_scenario.py:467]: Why a new test is added?

Comment (bebound) [reply src/connectedk8s/azext_connectedk8s/tests/latest/test_connectedk8s_scenario.py]: This might be introduced when Git merge conflict. Fixed.

**Review by jiasli — APPROVED** [review]

**Pull request merged**